### ISSUE TITLE
Openly transphobic command in .rategirl command

### ISSUE THREAD

**Issue by minakitty** [Kwoth/NadekoBot#2630]
I learned about this from another site, and was able to verify on both a server that I own and multiple servers that I'm an admin or member on that use this bot that it still exists.  It is specifically, this image: 
![screenshot_63](https://user-images.githubusercontent.com/5273418/47525848-1621d000-d853-11e8-8015-e1c060c13664.png)

I get that it's meant as a joke, likely referencing to "The Universal Hot vs Crazy Matrix" on YouTube, but the language of using "Tranny" and making them unrateable is highly offensive to many members of the transgender community.  Transgender people are, simply put, people.  Not some non-entity that deserve to be shamed and ridiculed.  And since people shouldn't be getting rated into that category anyway (because it's outside the chart), simply removing it from the base image file should be enough to clear up a number of issues

**Comment by Potato1680** [comment]
Wha Wha Wha Wha Wha. Do you hear that? It's the sound of someone crying about something they don't like. Well, do I have the solution for you! If you don't like it, then don't use it! WOW!

**Comment by Kaoticz** [comment]
If you don't like it, run `.sc ".rategirl" disable` on your Discord server.
The command will not be removed from the public bot, nor the image will be modified to suit your needs.

Alternatively, you can self-host your own Nadeko. You are free to modify whatever you wish on it.

**Comment by JapanYoshi** [comment]
I disagree with both Kaoticz and Potato1680. You're putting this bot out into the public. You wouldn't put a command that shows you the N-word every time in such a bot, unless you support racism. It's the same with the T-slur and transphobia. It's simple human decency.

**Comment by JapanYoshi** [comment]
![](https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSaN_DO2bk1UrNNWzqvowwM-DQzqha1buRxaHJ_J12YLYNdJmcT)

**Comment by Kaoticz** [comment]
I wasn't arguing. I didn't say the command is not going to get removed/modified because I didn't agree with the presented point, I said the command is not getting removed/modified, period.

Hence why I presented a solution.
This is GitHub, not Tumblr.

**Comment by JapanYoshi** [comment]
This isn't an argument. You may disagree with the point that your bot is needlessly transphobic. It is, and your opinion doesn't matter against the fact that your bot uses a pejorative.

Your solution is for us to stop using a part of your bot, but at this point, our solution is for us to stop using the bot, and telling people to stop using the bot, until you fix this. I'm not even telling you to apologize for the disrespect. This is GitHub, not 4chan.

It would take very little image editing to fix, yet you choose to use *way* more effort on ridiculing people who already suffer from discrimination (which you feed).

**Comment by Kaoticz** [comment]
The bot is not mine, nor do I have the power to `fix` it, as you have implied.
My solution stands. Use `.sc`, self-host and modify it yourself or stop using public Nadeko entirely.

**Comment by JapanYoshi** [comment]
Not a solution. Try again.

**Comment by Kaoticz** [comment]
This ticket can be closed.

**Comment by Potato1680** [comment]
Wait, don't close it yet. I want to make fun of the challenged person.

**Comment by Potato1680** [comment]
Trust me sweaty, I'm already dead inside. That's why I spend all my time on Discord. I'm fully aware that this is public and I hope the whole world gets to see this. I hope everyone knows how desperate you guys are getting that you start complaining on GitHub, of all things. You guys (girls?)(girlguys?)(guygirls?) are fighting a losing battle. Good luck in the real world, you are going to need it if you wish to get far in life.

**Comment by minakitty** [comment]
@JapanYoshi it's not something they're going to care about.  Even the person developing it has apparently said the point of the bot is to be offensive and unfair to everyone, and that's how they achieve equality.  That said, I haven't seen it mocking anything non 'PC' yet ... instead there's enough Pepe emoji's and such on their server that it's more likely they're interested in /pol/ than actual people.  If they choose to die on this hill, that's their choice.

**Comment by Potato1680** [comment]
Kindly, go spread your cancer else where. Thank you. @JapanYoshi  @minakitty

**Comment by JapanYoshi** [comment]
Here's a little story in closing:

In literature class, my teacher told us to make oxymorons.

“What's an oxymoron?” I asked.

“An oxymoron is a phrase that contradicts itself, like ‘loud silence’,” she said.

“Oh, like ‘good Republican’?” I replied.

**Comment by Potato1680** [comment]
Here's a little story in closing:

I could care less about politics.

I could care less about you.

I could care less about your group of nazis.

End of story.

**Comment by JapanYoshi** [comment]
You *could* care less, so you care enough about it to reduce the number of
fucks you give? Idiot.

**Comment by Potato1680** [comment]
You can't reduce 0. Unless we go into negatives, but that requires math and I don't like math. Kinda like how I don't like you and your friends.

**Comment by JapanYoshi** [comment]
But you said you “could care less”. If you could care less, you can't be
giving 0 fucks, because you *couldn't* care any less than that. Speak
proper English, you mouth-breathing cactus dildo.

2018年10月26日(金) 11:05 Potato1680 <notifications@github.com>:

**Comment by JapanYoshi** [comment]
P.S. You're the Nazi here. Think about it, actual Nazis put transgender people in concentration camps, not neckbeards.

**Comment by Potato1680** [comment]
"Speak proper English, you mouth-breathing cactus dildo." 

First off, I'm a potato. Please, call me by the proper term. Which would be 'Potato'.

Second, I guess this is the part where I get to call you names. Is it not? 

Well, lets start it off then. You are such a waste of an intelligent human being. (If I can even call you a human being at this point.)  You should also stop using your sexual preference as a lifestyle.  No one cares about what you are. Those who do care reserve no right. Just because something doesn't happen your way or because something isn't dealt with the way you want it to be dealt with, doesn't mean it's wrong or politically incorrect. It just means that other people have different opinions and views and that they handle things differently than you. That being said, if this is how you want to deal with a command on a Discord Bot then so be it. Your choice not mine. Also, being transgender is also a choice. It's not something you are born into.

P.S No one is putting you into a concentration camp. Just making fun of you and your kind.

**Comment by minakitty** [comment]
In fairness, I was having trouble finding where it was even coded (and therefore couldn't tell that it was a truly random thing).  But having people commenting that they are unable to do anything about it didn't contribute much to the conversation.  Admittedly, in general it did spiral quickly.

If someone would have pointed out where in the system it was, then a large portion of my issues would have been addressed.  I still don't like the image that's being used, but if I wanted to deal with the issue of self hosting, I could change the URI for the base image to something that whited out that section.

But at this point, I'd be more frustrated by the comments that seem to be coming from a point of belief here and on the discord server, telling transgender people that they're mentally ill, or making a choice (like someone would choose to be mocked and discriminated against just for the luls)

I get that Kwoth won't want to change anything, and even if I did a PR to make something a bit less sexist, it would likely get rejected.  I try to take a lot of stuff not seriously, but it really doesn't feel like some of the support staff are able to reconcile that 'transgender people are people too'

**Comment by vigilant-hylian** [comment]
I'm going to throw my hat into the ring here and say that it would be ethically more sound to remove this specific functionality of the game than to let it continue, now that it's been brought to light. The phrase 'tranny' is a slur to transgender groups an is unacceptable. The game itself is problematic but ultimately likely not be removed but something that is directly irresponsible like this slur needs to be removed.

Please consider doing the right thing, authors.

**Comment by FlirtySebastian** [comment]
It's fine as it is. it has nothing to do with trans. You're either a boy or a girl. sex change? ok good you changed to a different gender yay. But i don't see a reason to change this. delete issue and move on please.

**Comment by JapanYoshi** [comment]
Yippee, more transphobic drivel.

**Comment by zergling-man** [comment]
"now that it's been brought to light"
... As if it wasn't public knowledge? Anybody who has ever used the command knows it's there. And it's not like this is the first time we've had people bitching about it either. The answer has always been, and will always be, "we don't care, if you want to change it, host your own copy and change it". (More on this in my reply to the cat.)

But to fully nail this home: You think this thread is some sort of whistle-blowing. It isn't. It's just another wayward bitchfest that is surprisingly coming back to being on topic.

----

... I'm not sure how you thought it worked. Even if it weren't random* it'd be based on something equally irrelevant, like a hash of the username. (That's probably what I'd have done from the outset.)

You have also always had the opportunity to go look at the code yourself. Or ask where it is. (Though once you understand the base structure of modules and services, you don't really need to ask any more, just go digging and it appears exactly where you expect it to be.)

You have also always had the opportunity to host your own copy of the bot and change the matrix to whatever you like, including a version that is identical except without that area. It'd still be possible to roll into the blank space though, unless you also nuke those lines. You should spend some time learning about images.json and `.imagesreload`.

I'd like to point out that I am the person responsible for kicking the majority of the alts** that showed up yesterday, while laughing at it. I am still laughing at it. But not once have I done anything to indicate that transgendered people are less than human. I just find the entire premise of complaining about it hilariously inane.

*http://random.org it isn't *truly* random, by the way. RNG is a great subject.
**I'm sticking with this story on the hopes it'll aggravate the nutcases further.

**Comment by minakitty** [comment]
@zergling-man I figured it could have been some form of text scanning from previous posts, or something more complex than pure chance.  I tried looking for where it was in the code, but I probably would have made each module it's own file (that's my personal style, nothing against how it's done here, just like tabs vs. spaces)

Now that I know where the files are, if I chose to, I -could- easily change that game to ignore the section.  Or recode a new version called .rateguy (I might one day just to even things up)

As far as alts, I can't speak to that.  I have my main account, one alt account that I solely use for server testing, and is currently offline right now, and one bot (unrelated) that is down because of some authentication bug.  Thinking that I might be an alt of someone you kicked is your choice, it's just not true.

And by my comments about indicating that some on the Nadeko server act/imply that transgender people are less than human were in no way attributed to you.  You and Kaoticz have been the most helpful in this regard overall

**Comment by vigilant-hylian** [comment]
I'm going to link to this part of the ToS that this code is in breach of:

https://github.com/Kwoth/NadekoBot/blob/1.9/NadekoBot.Core/Modules/Games/Games.cs#L79, lines 128-133.

https://help.github.com/articles/github-terms-of-service/#c-acceptable-use

```
2. Content Restrictions
You agree that you will not under any circumstances upload, post, host, or transmit any content that:

is unlawful or promotes unlawful activities;
is or contains sexually obscene content;
is libelous, defamatory, or fraudulent;
**is discriminatory or abusive toward any individual or group;**
contains or installs any active malware or exploits, or uses our platform for exploit delivery (such as part of a command and control system); or
infringes on any proprietary right of any party, including patent, trademark, trade secret, copyright, right of publicity, or other rights.
```

Tranny is a phrase that, regardless of your feelings about transgender people, the LGBT community, politics etc, is a slur and derogatory to transgender people. It's not like calling someone an idiot, it's an actual slur used to oppress and hurt a specific group. That's not debatable. I don't care if you don't like transgender people, but the fact is that is is unacceptable content.

**Comment by zergling-man** [comment]
> Or the bit where it's only transphobic if you want it to be. (The saner among us have no problem reconciling "joke thing uses the word 'tranny'" with "transgendered people are people too".)

**Comment by vigilant-hylian** [comment]
That's exactly like using the n word and saying it's only offensive if you want it to be. A slur designed to hurt or damage or insult a group like this. There are slurs for all kinds of groups be they racial or sexual or gender or class and this is one about transgender people.

**Comment by Flatgub** [comment]
Perhaps at the very least it might be beneficial to rephrase parts of the command? Substitute the offending slur with a more preferable alternative in the image and rewrite the description. Its not an incredible solution sure, but amends the more egregious complaints without removing the command entirely (which seems to the the largest objection to any proposed changes) or cutting that section from the joke. 

Its a middle ground, the worst part of the command (the use of the slur and related implications) being fixed without the removal of the command. Unless of course the use of the slur is **absolutely crucial** to the entire joke and any changes made to it at all are unacceptable, in which case there's truly no hope.

**Comment by zergling-man** [comment]
No.

**Comment by JapanYoshi** [comment]
Good luck with your Discord Terms and Service violation, Kwoth's flunkies.

**Comment by zergling-man** [comment]
>Suddenly this person is of lesser value for being transgender, called a slur, and also called a 'dude' when by definition a trans woman is not recognised a woman except by bigots. The joke is the entire problem.
... Wait what. How do you even get that reading from it? (The original video doesn't elaborate any more than what's included in this implementation, so no hints from there.) I've always operated under the assumption that it actually refers to a trap (guy pretending to be a girl rather than actually transitioning/believing they are a girl), or an MTF transition.

---

>Thank you for the insightful criticism, you've clearly shown me the mistakes in my suggestion!
Glad you enjoyed it. 👌
To clarify:
>[Alternatively, you can self-host your own Nadeko. You are free to modify whatever you wish on it.](https://github.com/Kwoth/NadekoBot/issues/2630#issuecomment-433187243)
>Unless of course the use of the slur is absolutely crucial to the entire joke and any changes made to it at all are unacceptable
Yes and no, and yes and no.
It's crucial to accurately referencing the source, whether that person considers it crucial to their joke or not I don't know. I'd say yes since they saw the need to include it in the video. "any changes at all" are not unacceptable, but completely pointless changes (_this_) are.

**Comment by shivaco** [comment]
Imagine getting offended by a word "Tranny", though.

**Comment by Flatgub** [comment]
I appreciate your clarification.

Its already been somewhat established even by those who feel the command should stay that its not a particularly funny command, the joke isn't great in the first place, with or without the slur
>To be fair, `.rategirl` isn't even funny to begin with. But whatever, it is what it is.

Although I can't personally fathom anybody being so attached to this feature that its straying from the source material would upset them, other than *"It shouldn't be changed on principle"*, I'll give the benefit of the doubt and assume somebody does.

It should be noted that the joke in the first place is actually an expansion (albeit in somewhat poorer taste) of https://www.youtube.com/watch?v=2Eu4zwRcMM0 , which is the exact same joke in principle excluding more discreet partitioning of the acceptable side of the scale, and the inclusion of the section thats bothering anybody. Hell you could even partition this graph in a similar fashion.

You could feasibly replace `.rategirl`'s metric for this with the above material instead and maintain the same functionality without any issues of incorrect representation of the source material. Its still the same joke with a different source. 

I'll admit in all fairness that the only arguments I have against the solutions of 
>*"Just don't use the bot if you don't like it, or disable the feature"*

and 
>*"Host the bot yourself and remove it in your own build"* 

are that:

- Such content is potentially ToS breaking. The Discord Developer ToS states that neither you nor your application may break the Discord ToS, which as shown above the ToS prohibits the uploading/posting of content that is discriminatory, which the use of the slur in question inherently is, in the same way using the n-word to refer to people of colour as a group. While both words may, in singular or special case usages may be "acceptable", blanket usage of them to refer to a group of people is discriminatory.

- And in continuation of this, having such content in the first place is arguably distasteful. `.rategirl` is in essence no different than a command which would pick a random racial epithet and description for a person based on some arbitrary chart of "acceptable skin colours" or something of the like. Although perhaps such a command might be found similarly amusing to those who enjoy `.rategirl` in the first place, in which case all I can do is point towards the likely breach in ToS as still a valid cause for concern.

**Comment by Flatgub** [comment]
> Imagine getting offended by a word "Tranny", though.

People get offended by the word for the same reason people get offended by the n-word, they're both slurs. It's not some sort of neutral descriptor like the word transgender is.

**Comment by shivaco** [comment]
Would you report all the people who said "Tranny" or shit like that whether for shits and giggles or not? I bet you would.

**Comment by Flatgub** [comment]
What?
 For a start that's not even relevant, also no I wouldn't because that's a waste of time? I have other things to do than scouring the internet for people for people to report over single uses of the word? That's beyond petty.

**Comment by shivaco** [comment]
You can always kick the bot and never use it again. You can self-host and remove anything you don't fucking like. You can stop caring about shit and use the bot normally. Nobody is gonna do something just because you don't like it. Carry on.

**Comment by ne0ndrag0n** [comment]
I'll give my two cents on this. I am a transgender person. I'm personally flattered that, on this chart, I'm depicted as not only less crazy than the average woman but also at least a solid "8" on an attractiveness scale. I would personally disagree with this assessment that I can write my own ticket as far as partners go, not to mention that I've diagnosed myself with over a hundred different cancers due to anxiety, so I would disagree with the notion that I'm not certifiably insane as well. Nevertheless, I appreciate the compliment, so thank you for that.

The thing about language that people often forget is that context matters, and in the current run of open-source controversies regarding the use of alleged derogatory language, context is often ignored. I've personally used "tranny" informally, _not_ derogatorily, plenty of times. So have many of my friends, who are probably the farthest thing from easily-irritated, middle-aged Fox News watchers. I can see that its use here isn't intended to be derogatory. In fact, it's clearly comedic.

When I was growing up, the people who attacked comedy (as well as other forms of media) most often were social conservatives. They were a loud, obnoxious special interest group who claimed to reperesent the majority of America. Through organisations like the PMRC, the Moral Majority, and the PTC, they've waged character assasinations of anyone deemed "indecent", "blasphemous", or my personal favourite, "Satanic." On the rather questionable pretense that these groups had a broad mandate, many organisations caved into their demands. Lenny Bruce was repeatedly slammed with "obscenity" charges by puritannical local authorities. George Carlin was taken all the way to the Supreme Court for a routine discussing people's extreme reactions to swear words; the irony was certainly not lost on anyone paying attention.

Their tendrils extended into tech as well. Federated online services such as Prodigy were notorious for censorship, which only expedited their decline compared to decentralised alternatives such as local BBSs, Usenet, and eventually the Internet. Things we consider today, such as queer advocacy and frank sexual discussion, were _verboten_ on privately-operated platforms, who often billed themselves as being aggressively "family-friendly". By 1993, when their decline was irreversible, Prodigy was criticised by Wired for "the perils of trying to shape consumer behavior to a particular application of technology". 

Which is why I'm dismayed to see my compatriots, who often describe themselves as "liberal", now co-opting the _same tactics_ from the standpoint of social liberalism. If everything I'm talking about sounds familiar today, it's because it is. The only thing that has changed is the mandate, from an alleged conservative majority, to an alleged socialist majority, all while it's plainly obvious to anyone with both a job and a life that neither mandate has ever truly existed. And as a self-proclaimed liberal, I'm not inclined to suggest that these tactics are any more acceptable to adopt by our own side, than they were ever acceptable to adopt by our opposition.

Having said all that, I'll issue a personal recommendation, _not_ a demand, that the impact of this feature be reexamined. I still do believe it poses a moral hazard, but for different reasons. Personally, I find it odd that amid this controversy over use of the word "tranny", what's gone entirely ignored is the undertone that both women are chronically "crazy" and that their worth is measurable on a scale of physical attractiveness. Anyone who's had loved ones in their life who are girls/women, regardless of political affiliation, would find these insinuations unsavory and hurtful. Especially if they've known a loved one who's suffered anorexia, binge eating, and other image-based disorders.

In conclusion, It's ultimately the choice of the product maintainer(s), if they wish to retain this feature or not. The maintainer(s) free will to be distasteful is part of what makes a free society, free, and that's an important virtue that is slipping away from us as we speak. What is clear is that the feature is mildly distasteful, and that's far more of a problem than the offhanded use of a slur. 

To the OP, I thank you for your concern about transgender people like me. However, if you're going to weaponise rhethoric and outrage to punish those engaging in harmful behaviour, as opposed to engaging in constructive dialogue that furthers our goals and unites our country evermore, the least you can do is be honest about it.

**Comment by vigilant-hylian** [comment]
Shivaco, you're actually acting like a spoilt child right now. Taking cheap shots and reducing this conversation to personal attacks.

The fact of the matter is, it is unacceptable content, and you're being an ass.

In honesty, I am also transgender and I find it pretty frustrating that the term tranny is used in the way it is, and it's not debatable that it's a slur. Maybe some of us don't mind using it around each other, in a similar way that black people don't mind using the n word around each other but that doesn't stop it from being a slur. Period.

**Comment by Flatgub** [comment]
I don't think having a discussion about such things is a waste of time at all, if the discussion bothers you to such a passionate degree you don't have to take part? But regardless I already do self host a personal build and that's all I can really do, outside of partaking in this discussion about the topic.

**Comment by shivaco** [comment]
> Shivaco, you're actually acting like a spoilt child right now. Taking cheap shots and reducing this conversation to personal attacks.

> and you're being an ass.

Yes, I don't disagree on that one.


> but that doesn't stop it from being a slur.

Is it *really* supposed to be a slur in an output of a command that's been made just for fun?

> if the discussion bothers you to such a passionate degree you don't have to take part?

👌

**Comment by zergling-man** [comment]
That's one hell of a troll post. Good read. But uh:
>Personally, I find it odd that amid this controversy over use of the word "tranny", what's gone entirely ignored is the undertone that both women are chronically "crazy" and that their worth is measurable on a scale of physical attractiveness.

Meet:
>(Or the bit where the entire premise of choosing how to interact with women based on how hot and crazy they are is pretty sexist.)[https://github.com/Kwoth/NadekoBot/issues/2630#issuecomment-433265392]

---
Although I'm equally entertained by all this pussyfooting around "the n-word". Nigger is a word, some people hate it, but it's not Voldemort, saying it outright when discussing it isn't going to make Terry Crews suddenly appear and kill you by flexing his pecs. Using it offensively is not good, but not using it at all is... Honestly childish.
As much as I am here to mock idiots and laugh, I am also here to have a serious conversation with the adults willing to join me.
(Until it gets closed and locked like it should have been from the third post - which will be a bit after certain much more important matters get settled.)

**Comment by Flatgub** [comment]
Of course its a word and saying it wont have any real repercussions here, we're having a discussion around the topic, for anybody to think otherwise would be naive, however whether I chose to use the word or not doesn't change that you still understood the point being made, you understand the implication, of what word is being spoken about. Whether somebody prefers to use the word or not doesn't detract from your ability to understand and converse with them as you both know what each other means, some people just don't like to use such words, in the same way some people don't like to swear, that's a personal preference on their part.

What is childish on the other hand is to make insulting blanket accusations about those you're having a discussion against (potentially in bad faith, given you admit to being here to "mock and laugh") because you disagree with their decision on which words they want to use? I'm all for having a serious conversation but only on the grounds of it being actually legitimate and in good faith.

**Comment by KawaiiStalin** [comment]
If you don't like the command, simply disable it, or chose not to use it, easy peasy, problem solved. Can we please move on to real issues kthnx.

**Comment by minakitty** [comment]
@ne0ndrag0n I'm only saying something at this point because you called me out as OP.  I haven't mentioned before, because it shouldn't have been relevant to the conversation, but I'm also transgender, and know full well what it feels like to be marginalized.  But also understand that others have different values.  There's been some talk off site that I've been involved in and I'm looking at proposing a PR or two that may well mitigate some of the damage.  I won't presume that it's going to 'make everything all better', as some on one side want it to stay as is, and some on the other side want it gone entirely.  There's no way to make every one happy, and as PR's have to be merged by a developer, there's a small group of people that will make the decision for the whole project, no matter how much people may disagree with it.

They are right.  We can change it ourselves and self-host (an option which i've seen a number of people on the Nadeko discord use), or choose not to use or associate with servers that use the Nadeko bot.  Or we can attempt a compromise.  That's what I'm going to attempt to do.

**Comment by cloudrac3r** [comment]
Yes. This is a good PR.

**Comment by GithubIsAwful** [comment]
Jesus... GET A LIFE!

**Comment by cloudrac3r** [comment]
Alright, so you disagree with what I say. That's cool. But you could tell me *why* you disagree instead of applying a picture of your thumb.

**Comment by Midnight-Myth** [comment]
So, they should accept the PR and offend another group of people, just so that this specific group of people isn't offended anymore?
You know, people should be able to laugh about themself? Else they don't seem to be ok with being trans themself, if they can't even take a joke about it.

**Comment by cloudrac3r** [comment]
You know what? You're right. This command should be removed entirely in order to make everyone feel as good as possible.

I am 100% serious.

**Comment by JapanYoshi** [comment]
I'm fine with upsetting people by telling them they can't marginalize others.
